flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/FileArchivedExecutionGraphStore.java

```diff
@@ -102,6 +103,7 @@ public FileArchivedExecutionGraphStore(
 			"The storage directory must exist and be a directory.");
 		this.jobDetailsCache = CacheBuilder.newBuilder()
 			.expireAfterWrite(expirationTime.toMilliseconds(), TimeUnit.MILLISECONDS)
+			.maximumSize(maximumCapacity)
```

Comment:
Since the type of the origin parameter is `long`, we could change it accordingly.

Reply:
I think the int value range should be big enough. If this should become a problem we can change it later.